governance/policies/archiving.md

```diff
@@ -0,0 +1,102 @@
+
+## Policy
+
+### Compiler cutoff threshold
+
+The current compiler cutoff threshold is `4.08`.
+
+This threshold is subject to change by the opam repo maintainers based on the oldest compiler available on maintained distributions.
+
+<a name="inclusion-criteria"></a>
+### Criteria for inclusion to the primary repo (the default `opam-repository`)
+
+A version of a package may be published on the primary repo, and will continue to remain listed there, so long as the following criteria hold:
+
+1. The package meets the compiler cutoff threshold.
+2. The sources of the package version are available.
+3. The package version falls within a package's maintenance intent, or is a dependency of a package satisfying the primary repo criteria.
+4. The package version is installable on at least one of the supported platforms. (Note that it is not required that CI tests are passing, since working installation may require manual system configuration.)
+
+Note that this property is transitive along a package's dependency tree: if a package satisfies the primary repo criteria, then its dependencies do as well. 
+
+### Periodic pruning process
+
+At regular intervals, no less than every six months, the opam repo maintainers will reevaluate all packages to determine if they satisfy the [primary repo criteria](#criteria-for-inclusion-to-the-primary-repo-the-default-opam-repository). Packages that do not will be subject to archiving, according to the following process:
+
+- If the package version falls outside the package's maintenance intent, it will be archived.
+- The package version's maintainers will be notified of the intent to archive.
+  - Maintainers will have two weeks to fix the version so that it satisfies the criteria or approve of the archiving.
+  - If two passes without hearing from the maintainers, the package will be marked as unmaintained and a call for a new maintainer will be submitted to the community via [discuss.ocaml.org under the opam-repository topic](https://discuss.ocaml.org/tag/opam-repository).
+  - If a new maintainer steps forward, they will have 1 week to fix the package version.
+  - Otherwise, the package will be archived.
+
+
+### Archiving a package
+
+When it has been decided that a set of package versions (aka "versions") should be archived, archiving will proceed according to the following process:
+
+- A PR will be made to add the archived versions to the archive repo. 
+    - The opam file of each version will be extended as follows:
+        - Any dependencies without upper bounds will have upper bounds added, recording the latest available version of the dependency in the primary repo at the time of archiving.
+        -  The field `x-reason-for-archiving` will be added.
+        -  The field `x-opam-repository-commit-hash-at-time-of-archiving` will be added.
```

Comment:
This fieldname is quite the mouthful to keep track of...

Reply:
This will be done automatically via a custom tool in preparation, we can shorten the name but we don't expect that anybody will ever write it by hand in a opam file. It is also something that will only appear in the opam files in the other repo